azurerm/_modules/aks/main.tf

```diff
@@ -27,6 +27,19 @@ resource "azurerm_kubernetes_cluster" "current" {
 
     vm_size         = var.default_node_pool_vm_size
     os_disk_size_gb = var.default_node_pool_os_disk_size_gb
+    
+    vnet_subnet_id = var.vnet_subnet_id
+    max_pods       = var.max_pods
+  }
+
+  network_profile {
+    network_plugin      = var.network_plugin
+    network_policy      = var.network_plugin == "azure" ? "azure" : "calico"
+
+    docker_bridge_cidr  = "172.17.0.1/16"
+    service_cidr        = var.service_cidr
+    dns_service_ip      = var.dns_service_ip
+    pod_cidr            = var.network_plugin == "azure" ? null : "10.244.0.0/16"
```

Comment:
Is this the best place to "hardcode" these two CIDRs (docker_bridge_cidr and pod_cidr)? Or would these need to be configurable by the user?

Reply:
I believe so. For docker_bridge_cidr, the docs say it must be a range that is non-routable from the cluster, and by convention, should be set to 172.17.0.1/16.

For pod_cidr, it must be set to null if network_plugin = "azure", but could be user-configured if network_plugin = "kubenet". Going under the assumption that a user would want the default kubenet behavior if none of these keys are defined in config.auto.tfvars, the hard-coded values correspond to the defaults generated by AKS. It would be fairly easy to modify and allow a user-configured value for pod_cidr.